promgen/static/js/promgen.vue.js

```diff
@@ -180,7 +197,16 @@ app.component('silence-form', {
                     }
                 });
         },
-    }
+        hideModal: function (modalId) {
+            var modal = $('#' + modalId);
+            if (modal.length) {
+                globalStore.setMessages([]);
+                this.form = {};
+                this.state = silenceStore.state;
+                modal.modal('hide');
+            }
+        },
```

Comment:
```suggestion
        },
        showModal() {
            const modal = $('#silenceModal');
            if (modal.length) {
                // Detect when the modal is closed, and update the state accordingly. This is
                // necessary in case the user closes the modal by clicking outside of it, instead of
                // using the close button.
                //
                // https://getbootstrap.com/docs/3.3/javascript/#modals-events
                modal.on('hidden.bs.modal', function (e) {
                    silenceStore.hideModal();
                });
                modal.modal('show');
            }
        },
```

Reply:
This is the `showModal` method that we removed from the `app` component. Now it belongs to the `silence-modal` component.

When creating the Bootstrap modal, we add an event listener to it. The reason is explained in the comment above it.